linked_list/linked_list.py

```diff
@@ -1,75 +1,136 @@
 
 # Defines a node in the singly linked list
-class Node:
+from platform import node
+#from sys import last_value
 
+class Node:
     def __init__(self, value, next_node = None):
         self.value = value
         self.next = next_node
 
 # Defines the singly linked list
 class LinkedList:
     def __init__(self):
-      self.head = None # keep the head private. Not accessible outside this class
-
+        self.head = None # keep the head private. Not accessible outside this class
+       
     # returns the value in the first node
     # returns None if the list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
+    
     def get_first(self):
-        pass
-
+        if self.head == None:
+            return None
+        return self.head.value
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def add_first(self, value):
-        pass
+        self.head = Node(value, self.head)
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def search(self, value):
-        pass
+        current = self.head
+        while current != None:

```

Comment:
we can also shorten this to:
```suggestion
        while current:
```